hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AEndpointRegion.java

```diff
@@ -21,137 +21,138 @@
   @Test
-  public void testInvalidRegionDefaultEndpoint() throws Throwable {
-    describe("Create a client with an invalid region and the default endpoint");
+  public void testWithRegionConfig() throws IOException, URISyntaxException {
     Configuration conf = getConfiguration();
-    // we are making a big assumption about the timetable for AWS
-    // region rollout.
-    // if this test ever fails because this region now exists
-    // -congratulations!
-    conf.set(AWS_REGION, MARS_NORTH_2);
-    createMarsNorth2Client(conf);
+    conf.set(AWS_REGION, "us-east-2");
+
+    S3AFileSystem fs = new S3AFileSystem();
+    fs.initialize(new URI("s3a://landsat-pds"), conf);
+
+    Assertions.assertThat(fs.getInstrumentation().getCounterValue(STORE_REGION_PROBE))
+        .describedAs("Region is configured, region probe should not have been made").isEqualTo(0);
   }
 
   @Test
-  public void testUnsetRegionDefaultEndpoint() throws Throwable {
-    describe("Create a client with no region and the default endpoint");
+  public void testRegionCache() throws IOException, URISyntaxException {
     Configuration conf = getConfiguration();
     conf.unset(AWS_REGION);
-    createS3Client(conf, DEFAULT_ENDPOINT, AWS_S3_CENTRAL_REGION);
+    conf.unset("fs.s3a.bucket.landsat-pds.endpoint.region");
+    S3AFileSystem fs = new S3AFileSystem();
+
+    fs.initialize(new URI("s3a://landsat-pds"), conf);
+
+    Assertions.assertThat(fs.getInstrumentation().getCounterValue(STORE_REGION_PROBE))
+        .describedAs("Incorrect number of calls made to get bucket region").isEqualTo(1);
+
+    fs.initialize(new URI("s3a://landsat-pds"), conf);
+
+    // value should already be cached.
+    Assertions.assertThat(fs.getInstrumentation().getCounterValue(STORE_REGION_PROBE))
+        .describedAs("Incorrect number of calls made to get bucket region").isEqualTo(0);
   }
 
-  /**
-   * By setting the system property {@code "aws.region"} we can
-   * guarantee that the SDK region resolution chain will always succeed
-   * (and fast).
-   * Clearly there is no validation of the region during the build process.
-   */
   @Test
-  public void testBlankRegionTriggersSDKResolution() throws Throwable {
-    describe("Create a client with a blank region and the default endpoint."
-        + " This will trigger the SDK Resolution chain");
+  public void testEndpointOverride() throws Throwable {
+    describe("Create a client with no region and the default endpoint");
     Configuration conf = getConfiguration();
-    conf.set(AWS_REGION, "");
-    System.setProperty(AWS_REGION_SYSPROP, MARS_NORTH_2);
+
+    S3Client client = createS3Client(conf, AWS_ENDPOINT_TEST);
+
     try {
-      createMarsNorth2Client(conf);
-    } finally {
-      System.clearProperty(AWS_REGION_SYSPROP);
+      client.headBucket(HeadBucketRequest.builder().bucket(getFileSystem().getBucket()).build());
+    } catch (AwsServiceException exception) {
+      // Expected to be thrown by interceptor, do nothing.
```

Comment:
if you use intercept() an exception is raised if the exception isn't actually thrown. do you want that?